pybamm_tea/tea.py

```diff
@@ -303,14 +307,47 @@ def initialize(self):
+        if (
+            pava.get("Initial concentration in negative electrode [mol.m-3]") is not None
+            and pava.get("Initial concentration in positive electrode [mol.m-3]") is None
+        ):
+            raise ValueError("Error: 'Initial concentration in positive electrode [mol.m-3]' missing. Please, supply both or none.")
+        if pava.get("Initial loss of lithium inventory") is None:
+            pava["Initial loss of lithium inventory"] = 0
+            warnings.warn("Warning: 'Initial loss of lithium inventory' is set to 0.")
+        if (
+            pava.get("Initial concentration in negative electrode [mol.m-3]") is not None
+            and pava.get("Initial concentration in positive electrode [mol.m-3]") is not None
+        ):
+                    # calculate initial lithium inventory from initial concentrations in both electrodes and volume fractions and thicknesses
+                    pava["Initial lithium inventory [mA.h.cm-2]"] = (pava.get("Negative electrode thickness [m]") * pava.get("Negative electrode active material volume fraction") * pava.get("Initial concentration in negative electrode [mol.m-3]") + pava.get("Positive electrode thickness [m]") * pava.get("Positive electrode active material volume fraction") * pava.get("Initial concentration in positive electrode [mol.m-3]")) * 96485 / 3.6 / 10000
+                    pava["Initial lithium inventory / positive electrode theoretical capacity"] = pava.get("Initial lithium inventory [mA.h.cm-2]") / (pava.get("Maximum concentration in positive electrode [mol.m-3]") * pava.get("Positive electrode thickness [m]") * pava.get("Positive electrode active material volume fraction") * 96485 / 3.6 / 10000)
+                    pava["Initial loss of lithium inventory [mA.h.cm-2]"] = pava.get("Initial lithium inventory [mA.h.cm-2]") * pava.get("Initial loss of lithium inventory")
+                    pava["Lithium inventory [mA.h.cm-2]"] = pava.get("Initial lithium inventory [mA.h.cm-2]") - float(pava.get("Initial loss of lithium inventory [mA.h.cm-2]"))
+                    pava["Initial concentration in negative electrode [mol.m-3]"] = pava.get("Initial concentration in negative electrode [mol.m-3]") * (1 - pava.get("Initial loss of lithium inventory"))
+                    pava["Initial concentration in positive electrode [mol.m-3]"] = pava.get("Initial concentration in positive electrode [mol.m-3]") * (1 - pava.get("Initial loss of lithium inventory"))
+                    pava["Initial stoichiometry"] = pava.get("Lithium inventory [mA.h.cm-2]") / ((pava.get("Negative electrode thickness [m]") * pava.get("Negative electrode active material volume fraction") * pava.get("Maximum concentration in negative electrode [mol.m-3]") + pava.get("Positive electrode thickness [m]") * pava.get("Positive electrode active material volume fraction") * pava.get("Maximum concentration in positive electrode [mol.m-3]")) * 96485 / 3.6 / 10000)
```

Comment:
Is this what people normally mean be "initial stoichiometry"?

Reply:
I have no clue, but I don't think so, it is an artefact I delete now. (I don't remember exactly how I wanted to use that value.)